### PR TITLE
Use correct CSS version when injecting link tag

### PR DESCRIPTION
**Pull request by razor-x**
- refactor: Prefer disabled over enabled
- refactor: Return early from effect
- docs: Import CSS in example
- feat: Add useUnminifiedCss prop
- fix: Use correct version for stylesheet

### CODE REVIEW

File: examples/basic/src/App.tsx [hunk 1]
```diff
@@ -6,6 +6,7 @@ export const App = (): JSX.Element => {
       endpoint={import.meta.env.SEAM_ENDPOINT}
       publishableKey={import.meta.env.SEAM_PUBLISHABLE_KEY}
       userIdentifierKey={import.meta.env.SEAM_USER_IDENTIFIER_KEY}
+      disableCssInjection
```
Comment (razor-x) [inline examples/basic/src/App.tsx:9]: We have to put this in the example since when running locally the example should use the latest styles. We can keep the font injection.

File: package.json [hunk 2]
```diff
@@ -65,6 +65,7 @@
     "lint": "eslint --ignore-path .gitignore --ext .js,.cjs,.mjs,.ts,.tsx .",
     "prelint": "prettier --check --ignore-path .gitignore .",
     "postlint": "stylelint '**/*.scss'",
+    "prepack": "echo \"export default '$(jq -r .version < package.json)'\" > lib/version.js",
```
Comment (razor-x) [inline package.json:68]: Normally I would like to use a platform independent script (can always be done later), but this will ONLY run on CI so it should be safe to do this for now.

File: src/lib/seam/use-seam-styles.ts [hunk 3]
```diff
@@ -1,21 +1,31 @@
...
+}: {
+  unminified?: boolean
+  disabled?: boolean
+}) => {
+  const ext = `${unminified ? '' : 'min.'}css`
+  const cssUrl = `${origin}/v/${version ?? ''}/index.${ext}`
 
-export const useSeamStyles = ({ enabled }: { enabled: boolean }) => {
   useEffect(() => {
+    if (version === null) return
```
Comment (razor-x) [inline src/lib/seam/use-seam-styles.ts:18]: In development the version will always be null.

**Comment by razor-x** [comment]
UI test for the supported device table is flaky, seems it get stuck loading sometimes.

**Review by seveibar — APPROVED** [review]
fwiw i was testing this w/ `yalc` to make sure the imports go through properly, it's a bit hard to test, maybe you know a better way

File: src/lib/seam/use-seam-styles.ts [hunk 4]
```diff
@@ -1,21 +1,31 @@
 import { useEffect } from 'react'
 
-// TODO figure out the version automatically
-const cssUrl = 'https://www.unpkg.com/@seamapi/react@1.1.0/index.min.css'
+import version from 'lib/version.js'
+
+const origin = 'https://react.seam.co'
```
Comment (mikewuu) [inline src/lib/seam/use-seam-styles.ts:5]: If we always use the css file in the release / bundle, wouldn't it always have the correct version?

**Review by mikewuu — APPROVED** [review]

File: src/lib/seam/use-seam-styles.ts (hunk 4 above)
Comment (razor-x) [reply src/lib/seam/use-seam-styles.ts:5]: No because we are injecting a URL that contains the version number which must match the version of the package that is being used by the user. I'm not sure if that explanation is more clear, but it's correct.

**Comment by razor-x** [comment]
> fwiw i was testing this w/ yalc to make sure the imports go through properly, it's a bit hard to test, maybe you know a better way

@seveibar It's a little difficult to test things that only happen in the built package (very few things that hit that edge case). It's possible though: the `install` CI action actually ensures the package will be installable and the all it's modules will resolve.

This case it pretty special though. We could add an example app that ONLY uses the deployed version (would not "sync" with local dev). That would give a way to test it (I basically did that locally to verify this will work).

**Comment by razor-x** [comment]
@seveibar https://github.com/seamapi/react/issues/189

File: src/lib/seam/use-seam-styles.ts (hunk 4 above)
Comment (mikewuu) [reply src/lib/seam/use-seam-styles.ts:5]: > No because we are injecting a URL that contains the version number 

Yep, was wondering what the benefits of injecting via URL vs. just including what is bundled with the package are?

Comment (razor-x) [reply src/lib/seam/use-seam-styles.ts:5]: Ah, there is a pretty long internal discussion about this. The short version is that having it inject the fonts and the styles simplifies adoption and removes all the extra steps you need to use this. 

It's still possible to opt out of the HTML injection and do it the other way for users who prefer that.

Comment (mikewuu) [reply src/lib/seam/use-seam-styles.ts:5]: Was actually thinking of dynamically importing the css via relative paths, but of course that wouldn't work because this is runtime.

What if we always include the CSS, but just change the class name? Won't have to deal with version numbers, and no dynamic loading necessary?

```tsx
const class = disableStyles ? '' : 'seam-components'

return (<div className={class}>...</div>)
```

Comment (razor-x) [reply src/lib/seam/use-seam-styles.ts:5]: > What if we always include the CSS

I'm not sure I understand this. With this update we are always including the css unless the user opts out. I'd prefer not to inject the via an inline style tag since that bloats the js bundle size and cannot take advantage of caching the same way (among other things).

Comment (mikewuu) [reply src/lib/seam/use-seam-styles.ts:5]: >  I'd prefer not to inject the via an inline style tag

Yep, I mean we'd be able to avoid that. Here's a PR:

https://github.com/seamapi/react/pull/191
